SpringProject/src/it/java/com/gradproject2019/user/UserControllerIT.java

```diff
@@ -30,87 +37,146 @@
 
-        //when
         ResponseEntity<String> response = postUser(uri, request);
 
-        //Then
         Assert.assertEquals(400,response.getStatusCodeValue());
     }
 
     @Test
     public void shouldReturn400WhenUsernameWrongFormat() throws URISyntaxException {
-        //given
         URI uri = new URI(baseUrl);
         HttpEntity<UserRequestDto> request = new HttpEntity<>(createRequestDto("Karams Cool Username", "Karam", "Kapoor", "KSinghK@gmail.com", "wrong", "Botanist"));
 
-        //when
         ResponseEntity<String> response = postUser(uri, request);
 
-        //Then
         Assert.assertEquals(400,response.getStatusCodeValue());
     }
 
+    @Test
+    public void shouldReturn404WhenUserToBeEditedNotFound() throws URISyntaxException{
+        URI uri = new URI(baseUrl + "/1000000000");
+        HttpEntity<UserPatchRequestDto> request = new HttpEntity<>(createPatchRequestDto(null, null, null, null, null, null));
+
+        ResponseEntity<UserResponseDto> response = editUser(uri, request);
+
+        Assert.assertEquals(404, response.getStatusCodeValue());
+    }
+
+    @Test
+    public void shouldReturn200AndEditOnlyNotNullFields() throws URISyntaxException {
+        User user = new User ("Karams Cool Username", "Karam", "Kapoor", "KSinghK@gmail.com", "wrong", "Botanist");
+        User savedUser = userRepository.saveAndFlush(user);
+        URI uri = new URI(baseUrl + "/" + savedUser.getId());
+        String newUsername= "sophiaUsername";
+        String newFirstName= "Sophia";
+        HttpEntity<UserPatchRequestDto> request = new HttpEntity<>(createPatchRequestDto(newUsername,newFirstName,null, null, null, null));
+
+        ResponseEntity<UserResponseDto> response = editUser(uri, request);
+
+        Assert.assertEquals(200, response.getStatusCodeValue());
+        Assert.assertEquals(savedUser.getId(), response.getBody().getId());
+        Assert.assertEquals(newUsername, response.getBody().getUsername());
+        Assert.assertEquals(newFirstName, response.getBody().getFirstName());
+        Assert.assertEquals(savedUser.getLastName(), response.getBody().getLastName());
+        Assert.assertEquals(savedUser.getEmail(), response.getBody().getEmail());
+        Assert.assertEquals(savedUser.getOccupation(), response.getBody().getOccupation());
+    }
+
+    @Test
+    public void shouldReturn200WhenGettingUserById() throws URISyntaxException {
+        User user = new User( "Grace", "Grace", "Jones", "gbj@gmail.com", "P455w0rd!", "na");
+        User savedUser = userRepository.saveAndFlush(user);
+
+        Token testToken = new Token(savedUser.getId(), randomUUID());
+        Token savedToken = authRepository.saveAndFlush(testToken);
+
+        ResponseEntity<User> response = getUserById(savedUser.getId(), savedToken.getToken());
+
+        Assert.assertEquals(200, response.getStatusCodeValue());
```

Comment:
You should assert on at least some of the fields here, not just the response code